File: app/src/main/java/s15666/pjwstk/pamobmi/benedictharris/HarrisBenedictEquation.java
package s15666.pjwstk.pamobmi.benedictharris;

public abstract class HarrisBenedictEquation {

    public final double calculateEnergyExpenditure(Gender gender, EnergyExpenditure ee, double weight, double height, int age) {
        return ee.getMultiplier() * ( getGenderBaseValue(gender) + weight * getWeightMultiplier(gender)
                + height * getHeightMultiplier(gender) - age * getAgeMultiplier(gender));
    }

    abstract double getGenderBaseValue(Gender gender);
    abstract double getWeightMultiplier(Gender gender);
    abstract double getHeightMultiplier(Gender gender);
    abstract double getAgeMultiplier(Gender gender);
}

File: app/src/main/java/s15666/pjwstk/pamobmi/benedictharris/HarrisBenedictEquation.kt
package s15666.pjwstk.pamobmi.benedictharris

abstract class HarrisBenedictEquation {

    fun calculateEnergyExpenditure(gender: Gender?, ee: EnergyExpenditure, weight: Double, height: Double, age: Int): Double {
        return ee.getMultiplier() * (getGenderBaseValue(gender) + weight * getWeightMultiplier(gender) + height * getHeightMultiplier(gender) - age * getAgeMultiplier(gender))
    }

    abstract fun getGenderBaseValue(gender: Gender?): Double
    abstract fun getWeightMultiplier(gender: Gender?): Double
    abstract fun getHeightMultiplier(gender: Gender?): Double
    abstract fun getAgeMultiplier(gender: Gender?): Double
}